### PR TITLE
Add accessible download & media links

### PR DESCRIPTION
**Pull request by Mottie**
## Description

This PR adds 3 new components:

* `DownloadLink` - used for download links
    <details><summary>Provided examples</summary>
    
    
   **Download PDF Link Example**
    ```HTML
    <a 
      href="https://www.va.gov/playbook/downloads/Voices_Of_Veterans.pdf" 
      download="Voices_Of_Veterans.pdf" 
      type="application/pdf">
        <i aria-hidden="true" class="fas fa-download vads-u-padding-right--1" role="img"></i>
          Download the Voices of Veterans <dfn><abbr title="Portable Document Format">PDF</abbr> (4.9<abbr title="Megabytes">MB</abbr>)</dfn>
    </a>
    ```
    
    **Download Graphic Link Example**
    ```HTML
    <a 
      class="file-download-with-icon" 
      href="/img/2019-07/VA-020-Web-Badges-205x175-al-1.02.jpg" 
      download="/img/2019-07/VA-020-Web-Badges-205x175-al-1.02.jpg">
        <i aria-hidden="true" class="fas fa-download vads-u-padding-right--1" role="img"></i>
          Download VA benefits to people who've served <dfn><abbr title="graphic">JPG</abbr> (0.11<abbr title="Megabytes">MB</abbr>)</dfn> 
    </a>
    ```
    
    **Download Word Document Example**
    ```HTML
    <a 
      class="file-download-with-icon" 
      href="/files/2019-07/190111-ChooseVA-Marketing-Flyer-2019.docx" 
      download="/files/2019-07/190111-ChooseVA-Marketing-Flyer-2019.docx">
        <i aria-hidden="true" class="fas fa-download vads-u-padding-right--1" role="img"></i>
          Download Learn how to apply for VA benefits <dfn><abbr title="Word document">DOCX</abbr> (0.49<abbr title="Megabytes">MB</abbr>)</dfn>
    </a>
    ```
    </details>

* `MediaLink` - used for media (video & social media) links

    <details><summary>Provided example</summary>
    
    
    **View Video Example**
    ```HTML
    <a 
      class="video-link" 
      href="https://www.youtube.com/watch?time_continue=3&amp;v=xE0LzeRgdAk" 
      target="_blank" 
      rel="noopener ">
        <i aria-hidden="true" class="fas fa-play-circle vads-u-padding-right--1" role="img"></i>
          View video about VA disability compensation <dfn>on Youtube</dfn>
    </a> 
    ```
    </details>

* `AccessibleLink` - wrapper for `DownloadLink`, `MediaLink` & `Telephone`

### AccessibleLink

`AccessibleLink` combines the available link components. Only one component is then needed to render multiple link types.

```js
import AccessibleLink, {
  CONTACTS,    // Telephone contacts
  PATTERNS,    // Telephone pattern
  MEDIA_SITES, // MediaLink sites
  FILE_TYPES,  // DownloadLink known MIME-types
} from '@department-of-veterans-affairs/formation-react/AccessibleLink';

// Download minimal example
<AccessibleLink.Download
  href="https://example.va.gov/form.pdf"
  title="example form"
  size={1.2e6} // bytes, or pass in "1.2Mb"
/>

// Media link minimal example
<AccessibleLink.Media
  href="https://www.youtube.com/watch?v=xE0LzeRgdAk"
  title="VA disability compensation"
/>

// Telephone link minimal example
<AccessibleLink.Telephone contact={CONTACTS.VA_BENEFITS} />
```

<details><summary>Rendered HTML of above components</summary>

```html
<a
  href="https://example.va.gov/form.pdf"
  type="application/pdf"
  download
>
  <i role="img" aria-hidden="true" class="vads-u-padding-right--1 fas fa-download" />
  Download example form
  <dfn>
    <abbr title="Portable Document Format">PDF</abbr>
    (1.2<abbr title="Megabytes">MB</abbr>)
  </dfn>
</a>

<a
  href="https://www.youtube.com/watch?v=xE0LzeRgdAk"
  target="_blank"
  rel="noopener noreferrer"
>
  <i role="img" aria-hidden="true" class="vads-u-padding-right--1 fab fa-youtube"/>
  View video about VA disability compensation
  <dfn>on YouTube</dfn>
</a>

<a href="tel:+18008271000" aria-label="8 0 0. 8 2 7. 1 0 0 0." class="no-wrap">
  800-827-1000
</a>
```
</details>

### Full list of properties

<details><summary>DownloadLink</summary>

```js
/**
 * DownloadLink properties
 * @typedef DownloadLink~props
 * @type {object.<string, *>}
 * @property {string} href (required) - url pointing to a downloadable file
 * @property {string} title - title of the file; required if no children are passed
 *  into this component
 * @property {string|number} size - size of downloadable file; pass in string
 *  (number + unit abbreviation, e.g. "1.5mb" or "100kb") or a file size number
 *  in bytes.
 * @property {string} download - new download file name and extension of the
 *  downloadable file
 * @property {string} type - downloadable file MIME-type; include if the file is
 *  not in the list of known file extensions
 * @property {string|ReactElement} icon - Fontawesome icon group + name, e.g.
 *  "fas fa-download", or some custom icon JSX
 * @property {boolean} external - Set to true if the link is to an external site;
 *  setting this adds a blank target (opens a new tab) & no-referrer and opener
 *  rel attribute
 * @property {function} content - Function that allows custom rendering of the link
 *  content; This function returns JSX
 * @property {...*} var_args - Include any extra parameters you want to add to the
 *  download link, e.g. `className`, `onClick`, etc.
 */
```
</details>

<details><summary>MediaLink</summary>

```js
/**
 * MediaLink properties
 * @typedef MediaLink~props
 * @type {object.<string, *>}
 * @param {string} href (required) - url pointing to the video hosting site
 * @param {string} title (required) - title of the file or video
 * @param {function} content - Function that allows custom rendering of the link
 *  content; This function returns JSX
 * @param {string|ReactElement} icon - Fontawesome icon group + name, e.g.
 *  "fab fa-youtube", or the icon JSX
 * @param {boolean} external - Set to true if the link is to an external site;
 *  setting this adds a blank target (opens a new tab) & no-referrer and opener
 *  rel attribute
 * @param {...*} var_args - Include any extra parameters you want to add to the
 *  video link, e.g. `className`, etc.
 */
```
</details>

Here is an [online demo of the `DownloadLink` and `MediaLink` components](https://plnkr.co/edit/aOhgB5ZjPx8IJb53) - Use "Click to run" to render the page. Click on "index.js" to modify the links.

Related links:
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/11852
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/281
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/152#issuecomment-553127340

## Testing done

Local unit tests added

## Screenshots

N/A

## Acceptance criteria
- AccesibleLink
  - [x] Combines `DownloadLink`, `MediaLink` & `Telephone` components
  - [x] Passes all tests
  - [x] Documented
- DownloadLink
  - [x] Creates an accessible download link
  - [x] Allow customization within the confines of the component
  - [x] Passes all tests
  - [x] Documented
- MediaLink
  - [x] Creates an accessible media (video) link
  - [x] Allow customization within the confines of the component
  - [x] Passes all tests
  - [x] Documented

## Definition of done
- [ ] Changes have been tested in vets-website
- [ ] Changes have been tested in IE11, if applicable
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import AccessibleLink, {
+  CONTACTS,    // Telephone contacts
+  PATTERNS,    // Telephone pattern
+  MEDIA_SITES, // MediaLink sites
+  FILE_TYPES,  // DownloadLink known MIME-types
+  FILE_SIZES,  // DownloadLink known file sizes (B, KB, MB & GB)
+} from '@department-of-veterans-affairs/formation-react/AccessibleLink';
+
+// Download minimal example
+<AccessibleLink.Download
```
Comment (jbalboni) [inline packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:27]: This component's interface is pretty focused on the use case of a static PDF/file. Another common use case is a dynamically generated file or a file based on structured api data. That's mostly what I've had to deal with when building file links on va.gov and this interface doesn't work all that well for those cases, which would definitely benefit from the a11y work in here. Is it worth gathering some information on that use case before making this a formation component?

Comment (Mottie) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:21]: I tried to make this component flexible enough to work in most scenarios. Would you please share an example of how it would be difficult to use this component? I guess we can continue this discussion in slack, maybe in the #vetsgov-accessibility channel?

Comment (jenstrickland) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:21]: @jbalboni Yes, this is something I'm concerned about. The recent BAM2 Debt Letters use case and the download attribute issue with the Content-Disposition HTTP header's filename parameter — frankly some of it I don't yet understand completely — gave me pause about including the download attribute without some further exploration.  Do y'all or @rianfowler or @1Copenut have ideas on how we can support teams with this? Should we use this attribute? If yes, can we provide guidance for how to use it and the content-disposition header parameters ([review Slack thread](https://dsva.slack.com/archives/CBU0KDSB1/p1596751922192300?thread_ts=1596742928.177700&cid=CBU0KDSB1))?

Comment (jbalboni) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:21]: I'm not sure about the use case with the `download` attribute and a file from api. On VAOS we have a generated ICS file that uses the `download` attribute and handles IE separately: https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/vaos/components/AddToCalendar.jsx#L27. It might be taking on too much to incorporate that kind of logic in this component, I'm not really sure.

Comment (Mottie) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:21]: I've updated the components to now include a `content` callback function to render any custom link content. I've also updated the demo - https://plnkr.co/edit/aOhgB5ZjPx8IJb53; please let me know what you think.

Comment (jenstrickland) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:21]: I think the `A link to Download a huge legacy MS document file DOC (1.1GB).` example shows what I'm worried about. It didn't download. When I tested the BAM2 download letter pdf the browser hung. Maybe plnkr just shows the unhappy face instead?

Comment (Mottie) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:21]: > It didn't download

That example isn't using a real URL - https://example.va.gov/big-doc.doc - so I wouldn't expect it to work. The "Download this random json (blob) file" however will work; it's a "created" download link that includes content I created within that demo.

Comment (jenstrickland) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:21]: That link starts with `blob`, though, which doesn't replicate the issue with the BAM2 debt letters scenario that doesn't respect the download attribute, also seems to hang unless opened in a new tab. Would it be easier if we chatted about this on a call on Wednesday?

Comment (Mottie) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:21]: Yes, I would appreciate it.

File: packages/formation-react/src/components/AccessibleLink/AccessibleLink.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import { createIcon } from '../../helpers/link-utils';
+
+const AccessibleLink = props => props;
+
+AccessibleLink.Telephone = Telephone;
+AccessibleLink.Download = DownloadLink;
+AccessibleLink.Media = MediaLink;
+
+export { CONTACTS, PATTERNS, MEDIA_SITES, FILE_TYPES, createIcon };
+export default AccessibleLink;
```
Comment (ncksllvn) [inline packages/formation-react/src/components/AccessibleLink/AccessibleLink.jsx:13]: To fix the build, maybe we can try 

```
export default { Telephone, DownloadLink, MediaLink }
```

**Comment by Mottie** [comment]
YAY all checks are passing now 😅

File: packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx (hunk 1 above)
Comment (jenstrickland) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:41]: From our meeting last week regarding a download link using a dynamically generated file or a file based on structured api data, it was decided:

- The file name would be established by the backend dev in the Content-Disposition HTTP header's filename parameter using meta data, working with the designer and accessibility to ensure a meaningful filename.
- The `download` attribute would not be included in the link.
- It may be helpful to add target="_blank" to the download link if there is a delay in downloading the file, as it signals to the user that something is happening, as long as the new tab/window closes once the download is complete.

Comment (Mottie) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:41]: @jenstrickland The `download` attribute has 2 settings:
* `download` (equivalent to `download="true"` sets the link as a downloadable file; this is necessary to establish the link behavior
* `download="filename.ext"` sets the filename and extension of the downloaded file; but, as we discovered, not all browsers will use this setting.
* Currently, this component doesn't set these download links `target="_blank"` attribute, but it would if you passed in an `external` (`true`) property. Maybe I could make it `true` by default then?

Comment (jenstrickland) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:41]: We don't want a `target="_blank"` by default, only available as needed for cases such as the BAM2 generated file that seemed unresponsive while it generated the file for download.

Comment (Mottie) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx:41]: DOH, I just changed, and reverted, making download links use `target="_blank"` by default.

**Review by cvalarida — DISMISSED** [review]
Hey, @Mottie. This PR is too much to review at one time. VSP's [code review guidelines](https://github.com/department-of-veterans-affairs/va.gov-team/blob/277392988a09d59b244c04f54a7033822f371f1e/platform/engineering/code_review_guidelines.md#basic-expectations-for-pull-requests) states:
> changes should be smaller than a few hundred changed lines and a couple dozen files.

Please break this up into smaller PRs. I'd recommend one for each new component.

File: packages/formation-react/src/components/MediaLink/MediaLink.jsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+  const renderedSiteName = mediaSite.name && <dfn>on {mediaSite.name}</dfn>;
+  const renderedSiteType = mediaSite.type || 'site';
+  return (
+    <a {...linkProps}>
+      {typeof icon !== 'string'
+        ? icon
+        : createIcon(icon || mediaSite.icon || 'fas fa-play-circle')}
+      {(typeof content === 'function' &&
+        content({ renderedSiteName, renderedSiteType, linkProps, props })) || (
```
Comment (cvalarida) [inline packages/formation-react/src/components/MediaLink/MediaLink.jsx:100]: In what instance would the content need to be a React component that needs all these props? I think the more semantic way of using this component would be with `children`:
```jsx
<MediaLink href="https://www.example.com/link/to/video.mp4">Watch <em>this!</em></MediaLink>
```

File: packages/formation-react/src/components/MediaLink/MediaLink.jsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        <>
+          View {renderedSiteType} about {title} {renderedSiteName}
```
Comment (cvalarida) [inline packages/formation-react/src/components/MediaLink/MediaLink.jsx:102]: Does it make sense to have default text in here? It doesn't sit quite right with me, but I can't quite put my finger on why that is.

File: packages/formation-react/src/components/MediaLink/MediaLink.jsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+ * MediaLink component
+ * @function MediaLink
+ * @param {MediaLink~props}
+ */
+const MediaLink = props => {
+  if (!props.href) {
+    throw new Error('Media links require an href property');
+  }
+
+  const { external = true, title, content, icon = '', ...linkProps } = props;
```
Comment (cvalarida) [inline packages/formation-react/src/components/MediaLink/MediaLink.jsx:77]: Can we determine if `external` is true automatically by comparing the `hostname` of the `href` to `window.location.hostname`? It would be easy if IE supported the `URL()` constructor. :unamused: There might be a workaround there, though; I haven't looked.

File: packages/formation-react/src/components/DownloadLink/DownloadLink.jsx [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+
+  if (!linkProps.type) {
+    linkProps.type =
+      (FILE_TYPES[group] && FILE_TYPES[group][fileExt]) ||
+      'application/unknown';
+  }
+
+  const renderedSize = renderFileSize(size);
+  const renderedExtension = fileExt && (
+    <abbr title={group || fileExt}>{fileExt.toUpperCase()}</abbr>
```
Comment (cvalarida) [inline packages/formation-react/src/components/DownloadLink/DownloadLink.jsx:178]: `group` will never be falsey here if `getExtGroup` always returns either the known file type group or `Unknown`.

File: packages/formation-react/src/components/DownloadLink/DownloadLink.jsx [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+  );
+  return (
+    <a {...linkProps}>
+      {typeof icon !== 'string' ? icon : createIcon(icon || 'fas fa-download')}
+      {(typeof content === 'function' &&
+        content({ renderedExtension, renderedSize, linkProps, props })) || (
```
Comment (cvalarida) [inline packages/formation-react/src/components/DownloadLink/DownloadLink.jsx:184]: Same question as with `MediaLink`; can we use `children` instead?

File: packages/formation-react/src/components/DownloadLink/DownloadLink.jsx [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+        <>
+          Download {title}{' '}
```
Comment (cvalarida) [inline packages/formation-react/src/components/DownloadLink/DownloadLink.jsx:186]: Same question as with `MediaLink`; does it make sense to have default text in here?

File: packages/formation-react/src/components/DownloadLink/DownloadLink.jsx [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+          <dfn>
+            {renderedExtension} {renderedSize}
+          </dfn>
```
Comment (cvalarida) [inline packages/formation-react/src/components/DownloadLink/DownloadLink.jsx:189]: Same question as with `MediaLink`; should this be a definition element?

File: packages/formation-react/src/components/DownloadLink/DownloadLink.jsx [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+ * @typedef DownloadLink~props
+ * @type {object.<string, *>}
+ * @property {string} href (required) - url pointing to a downloadable file
+ * @property {string} title - title of the file; required if no children are passed
+ *  into this component
+ * @property {string|number} size - size of downloadable file; pass in string
+ *  (number + unit abbreviation, e.g. "1.5mb" or "100kb") or a file size number
+ *  in bytes.
+ * @property {string} download - new download file name and extension of the
+ *  downloadable file
```
Comment (cvalarida) [inline packages/formation-react/src/components/DownloadLink/DownloadLink.jsx:119]: What do you think about calling this `downloadFileName`?

File: packages/formation-react/src/components/DownloadLink/DownloadLink.jsx [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+ * @property {string} type - downloadable file MIME-type; include if the file is
+ *  not in the list of known file extensions
```
Comment (cvalarida) [inline packages/formation-react/src/components/DownloadLink/DownloadLink.jsx:121]: What do you think about calling this `mimetype`?

File: packages/formation-react/src/components/DownloadLink/DownloadLink.jsx [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+ * DownloadLink component
+ * @function DownloadLink
+ * @param {DownloadLink~props}
+ */
+const DownloadLink = props => {
+  if (!props.href) {
+    throw new Error('Download links require an href property');
+  }
+  const {
+    external = false,
```
Comment (cvalarida) [inline packages/formation-react/src/components/DownloadLink/DownloadLink.jsx:142]: Same as with `MediaLink`; can we determine this automatically?

File: packages/formation-react/src/components/DownloadLink/DownloadLink.unit.spec.jsx [hunk 13]
```diff
@@ -0,0 +1,348 @@
...
+  [1000, '(1000<abbr title="Bytes">B</abbr>)'],
+  [1050, '(1<abbr title="Kibibytes">KiB</abbr>)'],
+  [1170, '(1.1<abbr title="Kibibytes">KiB</abbr>)'],
+  [12345, '(12.1<abbr title="Kibibytes">KiB</abbr>)'],
+  [3.2e6, '(3.1<abbr title="Mebibytes">MiB</abbr>)'],
+  [4.57e9, '(4.3<abbr title="Gibibytes">GiB</abbr>)'],
+];
+
+describe('Widget <DownloadLink />', () => {
+  describe('utility functions', () => {
```
Comment (cvalarida) [inline packages/formation-react/src/components/DownloadLink/DownloadLink.unit.spec.jsx:74]: Can you move these tests to a different file? I'd also like to see tests clearly articulate the edge cases for each utility.

File: packages/formation-react/src/components/DownloadLink/DownloadLink.unit.spec.jsx [hunk 14]
```diff
@@ -0,0 +1,348 @@
...
+      const testCase = testData.pdf;
+      const wrapper = shallow(
+        <DownloadLink href={testCase.href} title="foo" />,
+      );
+      expect(wrapper.props().href).to.equal(testCase.href);
+      expect(wrapper.text().trim()).to.equal(`Download foo PDF`);
+      wrapper.unmount();
+    });
+    Object.keys(testData).forEach(ext => {
+      it(`should render ${ext.toUpperCase()} link`, () => {
```
Comment (cvalarida) [inline packages/formation-react/src/components/DownloadLink/DownloadLink.unit.spec.jsx:127]: I like this automating the creation of tests. Very meta. :smile:

File: packages/formation-react/src/components/AccessibleLink/AccessibleLink.unit.spec.js [hunk 15]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+
+import AccessibleLink from './AccessibleLink';
+import { createIcon } from '../../helpers/link-utils';
+
+describe('Widget <AccessibleLink />', () => {
+  describe('utility functions', () => {
```
Comment (cvalarida) [inline packages/formation-react/src/components/AccessibleLink/AccessibleLink.unit.spec.js:9]: Can you move this to another file? One for all the utility tests would be great.

**Review by cvalarida — COMMENTED** [review]
After seeing how much of this PR was comments, I went ahead and reviewed it anyhow. Thanks to @rianfowler for pointing that out. In the future, I'd definitely like to see smaller PRs as this still took me a long time to review.

That aside, it looks like there's still some work that needs to be done here. If you have any questions about the comments I left, please reach out to me in the Slacks!

File: packages/formation-react/src/components/AccessibleLink/AccessibleLink.jsx [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+    <a {...linkProps}>
+      {icon && typeof icon === 'string' ? createIcon(icon) : icon}
+      {children}
+    </a>
+  );
+};
+
+AccessibleLink.Telephone = Telephone;
+AccessibleLink.Download = DownloadLink;
+AccessibleLink.Media = MediaLink;
```
[the remark this answers is not in the document]
Comment (Mottie) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.jsx:20]: This pattern isn't used elsewhere, but I thought it might be nice to make HOC that gives you access to all link-type components. They are all sort of related. If you think this isn't necessary, I can remove it.

File: packages/formation-react/src/helpers/link-utils.js [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+  ['GB', 'Gigabytes', 'GiB', 'Gibibytes'],
+];
+
+/**
+ * Extract file size data from a mixed value of number plus size unit, e.g.
+ * "1.5mb"
+ * @param {string|number} size - file size, e.g. "1.5mb" or 1024 (bytes)
+ * @return {ReactElement} rendered size with unit abbr & full unit in title
+ */
+export const renderFileSize = (size = null, si = true, decimalPlaces = 1) => {
```
[the remark this answers is not in the document]
Comment (Mottie) [reply packages/formation-react/src/helpers/link-utils.js:65]: I don't know how I missed adding `si` and `decimalPlaces` to the JSDoc! 

I tend to avoid using external packages for small things because there is usually so much code-bloat... but maybe [Sindre's pretty-bytes](https://github.com/sindresorhus/pretty-bytes) package would be okay since it has no dependencies?

File: packages/formation-react/src/components/MediaLink/MediaLink.jsx [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+    );
+  }
+
+  // External site
+  if (external) {
+    linkProps.target = '_blank';
+    linkProps.rel = 'noopener noreferrer';
+  }
+
+  const renderedSiteName = mediaSite.name && <dfn>on {mediaSite.name}</dfn>;
```
[the remark this answers is not in the document]
Comment (Mottie) [reply packages/formation-react/src/components/MediaLink/MediaLink.jsx:92]: If you look at [Jennifer's original media link markup](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/152#issuecomment-553127340), she included the "on" inside the definition... I didn't think to question this part of it.

File: packages/formation-react/src/components/AccessibleLink/AccessibleLink.jsx (hunk 16 above)
[the remark this answers is not in the document]
Comment (cvalarida) [reply packages/formation-react/src/components/AccessibleLink/AccessibleLink.jsx:20]: Yeah, let's not add this new pattern yet.

File: packages/formation-react/src/helpers/link-utils.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (cvalarida) [reply packages/formation-react/src/helpers/link-utils.js:65]: I think if it's a well-supported dependency with a really narrow focus and won't add much to the end bundle size, I think adding an external package is fine. It looks like [pretty-bytes](https://www.npmjs.com/package/pretty-bytes) is quite popular and small, so I'd be :+1: to adding that if it solves the problem.

File: packages/formation-react/src/components/MediaLink/MediaLink.jsx (hunk 5 above)
Comment (cvalarida) [reply packages/formation-react/src/components/MediaLink/MediaLink.jsx:77]: :thinking: There might actually be a polyfill for the URL constructor. Not sure that's something we've got in our (outdated) `@babel/polyfill` import.

**Comment by bkjohnson** [comment]
@Mottie what's the status on this? Is work being done on it or can it be closed?

**Comment by Mottie** [comment]
I was planning on continuing work on this addition... I've just been overwhelmed with launching our HLR form.

**Comment by Mottie** [comment]
We can close this PR for now.. I can open a new one in the new repo you're working on.
